tests/integration_tests/bugs/test_lp1813396.py

```diff
@@ -0,0 +1,33 @@
+"""Integration test for lp-1813396
+
+Ensure gpg works even if VM provides no /dev/tty"""
+
+import pytest
+
+from tests.integration_tests.instances import IntegrationInstance
+from tests.integration_tests.log_utils import ordered_items_in_text
+
+
+USER_DATA = """\
+#cloud-config
+apt:
+  sources:
+    docker:
+      source: 'deb [arch=amd64] https://download.docker.com/linux/debian stretch stable'
+      keyserver: keyserver.ubuntu.com
+      keyid: 0EBFCD88
+"""  # noqa: E501
+
+
+@pytest.mark.sru_2020_11
+@pytest.mark.user_data(USER_DATA)
+def test_gpg_no_tty(client: IntegrationInstance):
+    log = client.read_from_file('/var/log/cloud-init.log')
+    to_verify = [
+        "Running command ['gpg', '--no-tty', "
+        "'--keyserver=keyserver.ubuntu.com', '--recv-keys', '0EBFCD88'] "
+        "with allowed return codes [0] (shell=False, capture=True)",
+        "Imported key '0EBFCD88' from keyserver 'keyserver.ubuntu.com'",
+        "finish: modules-config/config-apt-configure: SUCCESS",
+    ]
+    assert ordered_items_in_text(to_verify, log)
```

Comment:
This could also use a better failure message: now that we're using it in two places, perhaps `ordered_items_in_text` should return something (the first unfound item?) that could be used for a richer message? Or it could become `assert_ordered_items_in_text` and be responsible for raising an `AssertionError` with a richer message?

(A nit for this PR, but I would like to see this improved.)

Reply:
I guess I'm wondering what a different error message will buy us here. If it doesn't pass, we'll need to manually inspect the log file to see why, and I don't think a different error will change that.